src/main.ts

```diff
@@ -226,7 +277,11 @@ class AirQ extends utils.Adapter {
 		try{
 			const data = await this.getDataFromAirQ();
 			for (const element of this.sensorArray) {
-				this.setStateAsync(`Sensors.${element}`, { val: data[element][0], ack: true });
+				if(this.config.rawData){
+					this.checkNegativeValues(data, element) ? await this.setStateAsync(`Sensors.${element}`, { val: 0, ack: true }) : await this.setStateAsync(`Sensors.${element}`, { val: data[element][0], ack: true });
```

Comment:
const isNegative = this.checkNegativeValues(data, element)

[5,6,7]
data[element][0].map(
(v)=>isNegative ? 0 : v
)

Reply:
die map funktion geht hier so nicht, weil data[element][0] ein wert und kein array ist. schreibe es aber trotzdem etwas um. :)